include/vcpkg/base/checks.h

```diff
@@ -17,10 +18,17 @@ namespace vcpkg::Checks
         exit_with_message(line_info,
                           Strings::format(error_message_template, error_message_arg1, error_message_args...));
     }
+    template<class Message, class... Args>
+    [[noreturn]] typename Message::is_message_type exit_with_message(const LineInfo& line_info,
```

Comment:
Adding more layers that handle `Args&&...` is going to have code size costs; given how small the `msg::format` call is, I have a mild preference to just do this in the caller unless we are going to use this to enforce that all `exit_with_messages` are localized. (Since then we are getting a maintainability benefit for that code size)

Reply:
We are going to start using this to enforce all `exit_with_message` calls are localized, yeah.